### PR TITLE
Add algolia 📃🔍docsearch to kong docs

### PR DESCRIPTION
**Pull request by darrenjennings**
image

### CODE REVIEW

**Comment by darrenjennings** [comment]
[Add Algolia search to getkong.org](https://trello.com/c/KaQG4wym/674-add-algolia-search-to-getkongorg)

File: app/_assets/stylesheets/pages/docs.less [hunk 1]
```diff
@@ -21,8 +21,8 @@
         margin-top: 0.3rem;
         margin-bottom: 0.3rem;
       }
-    }
-
+    } 
+
```
Comment (thibaultcha) [inline app/_assets/stylesheets/pages/docs.less]: Let's trim those :)

File: app/_includes/anchor_links.html [hunk 2]
```diff
@@ -18,7 +18,6 @@
   };
   document.onreadystatechange = function () {
     if (this.readyState === "complete") {
-      console.log("here")
```
Comment (thibaultcha) [inline app/_includes/anchor_links.html]: :joy:

File: app/_layouts/docs.html [hunk 3]
```diff
@@ -80,10 +84,29 @@ <h5>{{nav.title}}</h5>
           <strong>Careful!</strong> You are browsing documentation for an outdated version of Kong. Go <a href="/docs/latest">here</a> to browse the documentation for the latest version.
         </div>
         {% endif %}
-
-      {{ content }}
+
```
Comment (thibaultcha) [inline app/_layouts/docs.html]: a few in here as well - intentional?

File: app/_layouts/docs.html [hunk 4]
```diff
@@ -80,10 +84,29 @@ <h5>{{nav.title}}</h5>
...
+<style>
+  @media (max-width: 768px) {
+    .algolia-docsearch-suggestion--subcategory-column{
+      display: none !important;
+    }
+  }
+</style>
+<script type="text/javascript" src="https://cdn.jsdelivr.net/npm/docsearch.js@2/dist/cdn/docsearch.min.js"></script>
+<script type="text/javascript"> docsearch({
+    apiKey: '691286ef280379144076a4f8df47a1d1',
```
Comment (thibaultcha) [inline app/_layouts/docs.html]: Just double checking... This is the search-only API key, right?

File: app/_layouts/docs.html (hunk 3 above)
Comment (darrenjennings) [reply app/_layouts/docs.html]: not intentional. will remove! 💅🏻

File: app/_layouts/docs.html (hunk 4 above)
Comment (darrenjennings) [reply app/_layouts/docs.html]: yes this is public facing search api key.

**Comment by thibaultcha** [comment]
@darrenjennings Just a few notes:

- the commit about fixing the anchor jumps could have been made in a separate PR - and be deployed already. We much favour atomicity and receiving multiple tiny, isolated contributions over a single large one.
- ditto for the linting commits
- the commit format of Kong/kong and Kong/getkong.org ideally does not uses a colon after the `type(scope)` prefix (waste of space). 

That said, no biggie, for next time :+1: (although it'd be nice to still fix the commit headlines)

**Comment by thibaultcha** [comment]
@darrenjennings We'll also need to fix the CI here before merging (you probably noticed already!)

**Comment by thibaultcha** [comment]
> Maybe squash and adjust the resulting commit message

Yes, squashing is very much required here!

**Comment by darrenjennings** [comment]
https://github.com/Kong/getkong.org/issues/71

**Pull request merged**